.github/workflows/tests.yml

```diff
@@ -19,13 +19,20 @@ jobs:
     timeout-minutes: 20
     steps:
     - name: Checkout source
-      uses: actions/checkout@93cb6efe18208431cddfb8368fd83d5badbf9bfd # v5.0.1
+      uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
     - name: Setup ruby
-      uses: ruby/setup-ruby@8aeb6ff8030dd539317f8e1769a044873b56ea71 # v1.268.0
+      uses: ruby/setup-ruby@d5126b9b3579e429dd52e51e68624dda2e05be25 # v1.267.0
       with:
         ruby-version: ${{ matrix.ruby }}
         rubygems: 3.4.10
         bundler-cache: true
+    - name: Enable virtual display & WebGL
+      run: |
+        sudo apt-get install -y xvfb mesa-utils libgl1-mesa-dri
+        Xvfb :99 -screen 0 1024x768x24 &
+        export DISPLAY=:99
+        echo "Using virtual display $DISPLAY"
+        glxinfo | grep "OpenGL renderer" || echo "GL info not available"
```

Comment:
These three lines are just debugging and should just be removed now you've got it working.

This step, which will then be just the one line to start `Xvfb` should be moved down just before the test steps that use it, and after the package install.